app/modules/claim.js

```diff
@@ -134,17 +150,48 @@ export const doGasClaim = () => async (
   }
 }
 
+export const checkClaimStatus = () => async (
+  dispatch: DispatchType,
+  getState: GetStateType
+) => {
+  const state = getState()
+  const net = getNetwork(state)
+  const txid = getTransactionId(state)
+  const endpoint = await api.loadBalance(api.getRPCEndpointFrom, { net })
+  const response = await rpc.Query.getRawTransaction(txid).execute(endpoint)
+  if (response.result && response.result.blockhash && response.result.confirmations) {
+    dispatch(setTransactionId(''))
+    dispatch(setFinalizeClaim(true))
```

Comment:
This is the only place we're calling `setFinalizeClaim`.  It seems like we need to call `setFinalizeClaim(false)` somewhere in order to reset it, otherwise only one claim per app launch will work (if I'm interpreting correctly).